raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -86,14 +126,15 @@ def estimate_gas(self, function: str, *args):
         try:
             return fn(*args).estimateGas({'from': to_checksum_address(self.jsonrpc_client.sender)})
         except ValueError as err:
-            tx_would_fail = (
-                '-32015' in str(err) or
-                '-32000' in str(err)
+            action = inspect_client_error(err, self.jsonrpc_client.eth_node)
+            will_fail = action in (
+                ClienErrorInspectResult.INSUFFICIENT_FUNDS,
+                ClienErrorInspectResult.ALWAYS_FAIL,
             )
-            if tx_would_fail:  # -32015 is parity and -32000 is geth
```

Comment:
what happened to the old error codes?

Reply:
They are wrong. As per the text of [this](https://github.com/raiden-network/raiden/pull/1858/commits/1a796bf5848fa1d66d75b281bbfcaa1e4e5832cf) commit message the only way to look this up is to look in the code of geth and parity.

And for example geth returns -32000 for multiple errors .. and forces us to check the error message. I complained a lot about this in rocketchat a few hours ago :(

Reply:
The previous logic was checking the if the code `-32015` was used by parity or if the code `-32000` was used by geth, on either `None` was returned.

The new logic is only checking the error code `-32010` from parity, and it's conditionally checking the error code `-32000` from geth.

I want to understand why parity's code `-32015` is not returning `None` anymore, and geth's `-32000` should raise more often.

[parity still uses the error code -32015](https://github.com/paritytech/parity/blob/21e0cd7781f55f0007d5c7d87929f49513367e27/rpc/src/v1/helpers/errors.rs#L38)

Reply:
That is a very good question for parity. I can not answer it. The reason I changed the error code is because the original issue which this PR fixes as reported by @pcppcp has the -32010 error code.  Check [here](https://github.com/raiden-network/raiden/issues/1815#issue-341052822)

`{'code': -32010, 'message': 'Insufficient funds. The account you tried to send transaction from does not have enough funds. Required 401482550000000000 and got: 374200502000000000.'}`

We could check for both `-32015` and `-32010` but then I want to see a real life example of `-32015` being used for insufficient funds and then I can do it.

> and geth's -32000 should raise more often

I don't get what you mean here.